src/sentry_utils.c

```diff
@@ -376,3 +376,43 @@ sentry__msec_time_to_iso8601(uint64_t time)
     strcat(buf, "Z");
     return sentry__string_clone(buf);
 }
+
+uint64_t
+sentry__iso8601_to_msec(const char *iso)
+{
+    int len = strlen(iso);
+    // The code is adapted from: https://stackoverflow.com/a/26896792
+    int y, M, d, h, m, s, msec = 0;
+    int consumed = 0;
+    if (sscanf(iso, "%d-%d-%dT%d:%d:%d%n", &y, &M, &d, &h, &m, &s, &consumed)
+        < 6) {
+        return 0;
+    }
+    // we must have a tail, either `Z` or starting with `.`
+    if (consumed >= len || (iso[consumed] != 'Z' && iso[consumed] != '.')) {
+        return 0;
+    }
+    if (iso[consumed] == '.') {
+        if (sscanf(&iso[consumed], ".%dZ", &msec) < 1) {
+            return 0;
+        }
+    }
+
+    struct tm tm;
+    tm.tm_year = y - 1900;
+    tm.tm_mon = M - 1;
+    tm.tm_mday = d;
+    tm.tm_hour = h;
+    tm.tm_min = m;
+    tm.tm_sec = s;
+    // a negative value means `mktime` should infer it
+    tm.tm_isdst = -1;
+    time_t time = mktime(&tm);
+    if (time == -1) {
+        return 0;
+    }
+    // add the UTC offset, since mktime assumes local time
+    time += tm.tm_gmtoff;
```

Comment:
`tm_gmtoff` is a nonstandard extension and not available in MSVC.

Reply:
Looks like on MSVC this should call [`_mkgmtime`](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/mkgmtime-mkgmtime32-mkgmtime64?view=vs-2019).

Reply:
There's a similar check in the function above. I believe this should check for the compiler rather than the plaform.

Reply:
You could probably do this (substituting with a check for the compiler instead):

```c
#ifdef SENTRY_PLATFORM_WINDOWS
time_t timestamp = _mkgmtime(&tm);
#else
time_t timestamp = mktime(&tm) - _timezone;
#endif
```

Reply:
> I believe this should check for the compiler rather than the plaform.

I think this rather depends on the platform libc, or does it?